### PR TITLE
Add empty states on invalid identifier in URL

### PR DESCRIPTION
**Pull request by fhlavac**
Fixes: https://issues.redhat.com/browse/RHCLOUD-11538

added empty states showing on invalid identifier in URL to:
- groups
- group roles
- group members
- roles
- users

image01
image02
image03

@mmenestr

### CODE REVIEW

**Comment by mmenestr** [comment]
Empty state texts look good, only thing is that it shouldn't be inside the page header, it should be in the page body! 
So the empty state should be on a grey background, while the breadcrumbs should be in the white page header!

**Comment by fhlavac** [comment]
No problem, I'll change it :-)

**Review by john-dupuy — APPROVED** [review]
Tested out locally and functionally this works great, I will add some test cases for these in the near future. 

Thanks @fhlavac!

File: src/test/smart-components/group/principal/__snapshots__/principals.test.js.snap [hunk 1]
```diff
@@ -194,18 +194,7 @@ Array [
...
-    Array [
-      "Add a user to configure user access.",
-      "",
-    ]
-  }
-  icon={[Function]}
-  title="There are no members in this group"
-/>
-`;
+exports[`<GroupPrincipals /> should render correctly with empty members 1`] = `null`;
```
Comment (karelhala) [inline src/test/smart-components/group/principal/__snapshots__/principals.test.js.snap:197]: Is this change correct? I can't see changes to `GroupPrincipals` and this snapshot has changed.

Comment (fhlavac) [reply src/test/smart-components/group/principal/__snapshots__/principals.test.js.snap]: No, it isn't, good catch, thank you. There was an outdated EmptyState component name in the expect statement.

**Comment by fhlavac** [comment]
@mmenestr no problem, I've updated the screenshots. What do you think about titles in the header? I think it's better than leaving only breadcrumbs there.

**Comment by fhlavac** [comment]
@mmenestr ok, it should be fixed. Updated the snapshots.

File: src/smart-components/group/group.js [hunk 2]
```diff
@@ -25,21 +26,26 @@ const Group = ({
...
+  const history = useHistory();
+
+  const breadcrumbsList = () => [
+    { title: 'Groups', to: '/groups' },
+    groupExists ? { title: isFetching ? undefined : group.name, isActive: true } : { title: 'Invalid group', isActive: true },
+  ];
 
   const fetchData = (apiProps) => {
-    fetchGroup(apiProps);
+    fetchGroup(apiProps).then(({ value }) => {
```
Comment (PanSpagetka) [inline src/smart-components/group/group.js:46]: I think it would be better to handle this in the reducer. Same for roles and users.

**Review by PanSpagetka — CHANGES_REQUESTED** [review]

**Comment by mmenestr** [comment]
Looks perfect thank you!!

File: src/smart-components/group/group.js (hunk 2 above)
Comment (fhlavac) [reply src/smart-components/group/group.js]: Moved the error flag to reducers, much better. Thanks :-)

**Review by PanSpagetka — APPROVED** [review]

**Pull request merged**